modelci/hub/profiler.py

```diff
@@ -47,16 +51,28 @@ def __init__(self, model_bo: ModelBO, server_name: str, inspector: BaseModelInsp
         self.model_info = model_bo
         self.docker_client = docker.from_env()
 
-    def diagnose(self, batch_size=None, device='cuda') -> DynamicProfileResultBO:
-        """Start diagnosing and profiling model."""
-
-        try:  # to check the container has started successfully or not.
-            self.docker_client.containers.get(self.server_name)
-        except Exception:
-            print(
-                '\n'
-                'ModelCI Error: starting the serving engine failed, please start the Docker container manually. \n'
-            )
+    def diagnose(self, batch_size: int = None, device='cuda', timeout=10) -> DynamicProfileResultBO:
```

Comment:
device = "cuda"

seems there is no clear way for CPU?

what if profiling models on CPU

Reply:
Current profiler fails in profiling on CPU @huangyz0918:
https://github.com/YuanmingLeee/ML-Model-CI/blob/0cb19ee10d05666f07fc4d3197e5bd0ada61f9b5/modelci/metrics/benchmark/metric.py#L132
This line fails, as there is no key named `'accelerators'` on CPU. Please help to fix this after this merge.

Reply:
I know this issue, it comes from cadvisor. Will fix, not hurry.

Reply:
Remember do not increase the scope of a current working pull request. Try to finalize the required changes ASAP.

Reply:
add a todo better
to remind of us and developers

Reply:
Yuanming has filed an issue for that @HuaizhengZhang